From ddc295a04269437e5c726f63239689786354d44b Mon Sep 17 00:00:00 2001
From: Sergey Matyukevich <geomatsi@gmail.com>
Date: Sat, 20 May 2017 11:24:26 +0300
Subject: [PATCH] ARM: dts: orange-pi-zero: enable SPI NOR

Enable SPI NOR on orange-pi-zero board.

For more information see:
- http://linux-sunxi.org/Orange_Pi_Zero#SPI_NOR_flash

Signed-off-by: Sergey Matyukevich <geomatsi@gmail.com>
[Dario: make the patch to be applied with fuzz factor 0]
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>
---
 .../boot/dts/allwinner/sun8i-h2-plus-orangepi-zero.dts    | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/allwinner/sun8i-h2-plus-orangepi-zero.dts b/arch/arm/boot/dts/allwinner/sun8i-h2-plus-orangepi-zero.dts
index 6b74cc923e16..4012f8712cd7 100644
--- a/arch/arm/boot/dts/allwinner/sun8i-h2-plus-orangepi-zero.dts
+++ b/arch/arm/boot/dts/allwinner/sun8i-h2-plus-orangepi-zero.dts
@@ -167,14 +167,18 @@ &ohci1 {
 
 &spi0 {
 	/* Disable SPI NOR by default: it optional on Orange Pi Zero boards */
-	status = "disabled";
+	status = "okay";
 
 	flash@0 {
 		#address-cells = <1>;
 		#size-cells = <1>;
-		compatible = "mxicy,mx25l1606e", "jedec,spi-nor";
+		compatible = "mxicy,mx25l1606e", "winbond,w25q128";
 		reg = <0>;
 		spi-max-frequency = <40000000>;
+
+		partition@00000000 {
+			reg = <0x00000000 0x200000>;	/* 2Mb */
+		};
 	};
 };
 
-- 
2.43.0

